indexer/services/comlink/src/controllers/api/v4/fills-controller.ts

```diff
@@ -94,6 +100,71 @@ class FillsController extends Controller {
       }),
     };
   }
+
+  @Get('/parentSubaccount')
+  async getFillsForParentSubaccount(
+    @Query() address: string,
+      @Query() parentSubaccountNumber: number,
+      @Query() market?: string,
+      @Query() marketType?: MarketType,
+      @Query() limit?: number,
+      @Query() createdBeforeOrAtHeight?: number,
+      @Query() createdBeforeOrAt?: IsoString,
+  ): Promise<FillResponse> {
+    // TODO(DEC-656): Change to using a cache of markets in Redis similar to Librarian instead of
+    // querying the DB.
+    let clobPairId: string | undefined;
+    if (isDefined(market) && isDefined(marketType)) {
+      clobPairId = await getClobPairId(market!, marketType!);
+
+      if (clobPairId === undefined) {
+        throw new NotFoundError(`${market} not found in markets of type ${marketType}`);
+      }
+    }
+
+    // Get subaccountIds for all child subaccounts of the parent subaccount
+    // Create a record of subaccountId to subaccount number
+    const childIdtoSubaccountNumber: Record<string, number> = {};
+    getChildSubaccountNums(parentSubaccountNumber).forEach(
+      (subaccountNum: number) => {
+        childIdtoSubaccountNumber[SubaccountTable.uuid(address, subaccountNum)] = subaccountNum;
+      },
+    );
+    const subaccountIds: string[] = Object.keys(childIdtoSubaccountNumber);
```

Comment:
nit: seems like we'll use this logic quite a bit, lets move it into a helper and reuse

Reply:
This is just a one-liner. We need both the subaccountIds and above map in the implementation below